### ISSUE TITLE
Default hxml when opening a project

### ISSUE THREAD

**Issue by ncannasse** [vshaxe/vshaxe#269]
I often have issue when opening a project for the first time: if there's several different hxml, the selected one is sometimes not the good one, but one used for "common" definitions.

I suggest the following way to resolve the default hxml:
- among the hxml which name starts with the same name as the workspace directory (case insensitive), use the first one
- otherwise, use current behavior (alphabetic order?)

Example:

```
/MyProject
  |- a.hxml
  |- myproject_hl.hxml
  |- myproject_js.hxml
  |- z.hxml
```

Should use `myproject_hl.hxml` by default

**Comment by Gama11** [comment]
I assume this if for Heaps samples? If there's any `haxe.displayConfigurations` defined, the first one will be picked for completion instead of the first auto-detected one. So you could generate a `settings.json` with this:

```json
 "haxe.displayConfigurations": [
    ["myproject_hl.hxml"]
]
```

I don't think I've really seen hxml files prefixed with the project name anywhere, the most common convention seems to be `build.hxml` or `build<Target>.hxml`.

**Comment by ncannasse** [comment]
It's not very heaps specific, just we have several projects at shiro and we often name the hxml with the name of the project, and we have some other hxml such as common.hxml etc. you're welcome to add also "build" prefix into the priority list.

**Comment by Gama11** [comment]
I'm not sure, that seems pretty arbitrary... Isn't it better to explicitly specify it in `settings.json` anyway? That way it's guaruanteed to work.

@Simn @nadako Any opinions on this?

**Comment by nadako** [comment]
Yes, I don't think we should add any magic here. Let's maybe re-visit this when our code is super-clean and we can afford adding more complexity :)

PS I think it's possible to write a custom configuration provider that would implement project/company specific convenience magic.

**Comment by Gama11** [comment]
>Let's maybe re-visit this when our code is super-clean

It's not? ;)

**Comment by RealyUniqueName** [comment]
Would be great if vscode ask the user which one of the discovered hxmls to use when opened for the first time.

**Comment by Gama11** [comment]
Hm, that's an interesting idea. We could do that only if there's multiple, and then also ask whether to save the choice in `settings.json` via `displayConfigurations`.

**Comment by ncannasse** [comment]
I have another version of my proposal : select the first hxml that has an actual output ( -js / -hl etc. ) , other hxmls are most likely parameters only.

**Comment by Gama11** [comment]
What's wrong with just explicitly asking the user which HXML to use when opening the project for the first time?
